shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/presentation/StepQuizReducer.kt

```diff
@@ -619,4 +640,25 @@ internal class StepQuizReducer(
                 )
             )
         )
+
+    private fun handleSeeHintsClicked(state: State): StepQuizReducerResult {
+        val step = (state.stepQuizState as? StepQuizState.AttemptLoaded)?.step
+        return if (step != null && StepQuizHintsFeature.isHintsFeatureAvailable(step)) {
+            val (newState, actions) = stepQuizChildFeatureReducer.reduce(
+                state = state,
+                message = Message.StepQuizHintsMessage(
+                    StepQuizHintsFeature.Message.LoadHintButtonClicked
+                )
+            )
+            newState to actions + Action.ViewAction.ScrollToHints
+        } else {
+            state to emptySet()
+        }
+    }
+
+    private fun handleReadCommentsClicked(state: State): StepQuizReducerResult =
+        state to setOf(Action.ViewAction.ShowComments)
+
+    private fun handleSkipClicked(state: State): StepQuizReducerResult =
+        state to setOf(Action.ViewAction.SkipStep)
```

Comment:
We also need to log analytic events when there is a click on the "see hint," "comments," and "skip" buttons from the feedback. It should be possible to distinguish between these clicks (toolbar and feedback) at the analytics level.

Reply:
@ivan-magda 
I fixed it.
Add separate messages for hint, comment & skip request, to not log click analytic event in the origine feature. And log required messages in the StepQuizReducer.

Reply:
Thanks 👍